### ISSUE TITLE
Allow listening for MDNS on a single interface instead of all interfaces

### ISSUE THREAD

**Issue by fbraz50** [lathiat/avahi#168]
It would be nice to be able to configure the server so that it listens only on a specific Interface and have it relay that data to all other interfaces vs. having the server listen on every single network interface and forward everything to all other interfaces.

Having the server relay everything everywhere causes drastic WiFi performance issues in larger environments due to all the broadcast traffic.

**Comment by allenzt** [comment]
you could try option  "allow-interfaces" in avahi-daemon.conf

>       allow-interfaces= Set a comma separated list of allowed network  inter‐
       faces  that should be used by the avahi-daemon. Traffic on other inter‐
       faces will be ignored. If set to an empty  list  all  local  interfaces
       except loopback and point-to-point will be used.

**Comment by BobSammers** [comment]
I would also like this. I have been trying to create a Docker container that advertises its own services via mDNS, but in order to do this in the normal Docker bridge mode, Docker needs to bind to udp 5353 on the host which it can't unless Avahi on the host is disabled.

I created a virtual interface using macvlan on the host with the intention of dedicating it to the container. Docker can handle bridging to a specific IP address fine, but because Avahi binds on 0.0.0.0 and, as far as I know, cannot be made to advertise only on one interface or ip address, it blocks binding on the new interface as well.

(I'm aware of the `allow-interfaces` and `deny-interfaces` settings in the `conf` file. They don't appear to affect the listener binding.)

**Comment by dereitz** [comment]
> I would also like this. I have been trying to create a Docker container that advertises its own services via mDNS, but in order to do this in the normal Docker bridge mode, Docker needs to bind to udp 5353 on the host which it can't unless Avahi on the host is disabled.
> 
> I created a virtual interface using macvlan on the host with the intention of dedicating it to the container. Docker can handle bridging to a specific IP address fine, but because Avahi binds on 0.0.0.0 and, as far as I know, cannot be made to advertise only on one interface or ip address, it blocks binding on the new interface as well.
> 
> (I'm aware of the `allow-interfaces` and `deny-interfaces` settings in the `conf` file. They don't appear to affect the listener binding.)

Did you ever find a workaround for this?

**Comment by BobSammers** [comment]
> Did you ever find a workaround for this?

No. The container in question is running in host mode, which is not ideal but avoids the need for Docker to have access to the mDNS port at all.

**Comment by ghost** [comment]
8 years for nothing? Gr8 https://github.com/avahi/avahi/issues/889

**Comment by fbraz50** [comment]
Wow, I posted this in 2018? My company at the time moved on. The idea was to rebroadcast MDNS signals from a source interface to other attached interfaces or networks for example:
- Interface 1 is on 192.168.10.0/24 network and this is where all of my MDNS devices live. Apple TVs/Airprint/chromecast devices.
- Interface 2 is on 192.168.11.0/24 one of my wifi network
- Interface 3 is on 192.168.12.0/24 a second wifi network

I was hoping to have interface 1 act as the "intake interface" as in the network that contains all of my broadcasting devices and rebroadcast those devices TO interfaces 2/3.

That was the dream at least.

Either way that company moved on and implemented Aruba wireless which has the MDNS rebroadcast capability built in.
